src/nrnpython/CMakeLists.txt

```diff
@@ -49,11 +49,18 @@ if(NRN_ENABLE_PYTHON_DYNAMIC)
     list(GET NRN_PYTHON_VERSIONS ${val} pyver)
     list(GET NRN_PYTHON_INCLUDES ${val} pyinc)
     list(GET NRN_PYTHON_LIBRARIES ${val} pylib)
+    set(nanobind_target "nanobind_py${pyver}")
+    make_nanobind_target(${nanobind_target} ${pyinc})
     add_library(nrnpython${pyver} SHARED ${NRNPYTHON_FILES_LIST})
+    if(MINGW)
```

Comment:
Could this be `_WIN32`? Or is it really specific to `MINGW`?

Reply:
Another explanation here:
https://cmake.org/cmake/help/latest/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html create a `definitions` file with all `.obj` used to create this target.
We can read `This property is implemented only for MS-compatible tools on Windows.`.
Internally in `CMake` they use the variable `CMAKE_SUPPORT_WINDOWS_EXPORT_ALL_SYMBOLS`, that is by default `0` and set to `1` on the compatible `Platforms`: `MSVC`, `PGI` and `Clang`. `MINGW` is not a `Platform` for `CMake` but `GNU` is, and this is not considered as a compatible one. So we need to set this _internal_ variable ourself.

So if you follow the stream, this is really specific to `MINGW`.

I can split this code in 2:
```
if(MINGW)
  set(CMAKE_SUPPORT_WINDOWS_EXPORT_ALL_SYMBOLS 1)
endif()
set_property(TARGET nrnpython${pyver} PROPERTY WINDOWS_EXPORT_ALL_SYMBOLS ON)
```
The second one is not a problem to be set everywhere (as this is disable by the above-mentionned variable).
If we want to be more clear, on the fact this is useful only for `WIN32` platform we can specify it.

Reply:
By the way in `CMake` it seems to be `WIN32` and not `_WIN32`

Reply:
done